enhancements/dns/baremetal-optional-mdns.md

```diff
@@ -0,0 +1,162 @@
+---
+title: baremetal-optional-mdns
+authors:
+  - "@cybertron"
+reviewers:
+  - "@celebdor"
+approvers:
+  - TBD
+creation-date: 2019-11-21
+last-updated: 2019-11-21
+status: implementable
+---
+
+# Baremetal Optional mDNS
+
+Make deployment of the baremetal mDNS services optional to support environments
+where multicast is not allowed and alternate solutions for providing internal
+DNS records are used.
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+1. This may be problematic to test because it requires an external provider
+   for DNS records. It's possible we can add support for creating those
+   records in openshift-metal3/dev-scripts.
+1. It's not clear at this time exactly what combination of mDNS services
+   will be required. We may only be able to remove mDNS from the workers, but
+   still need it on the masters. The current design is flexible enough to
+   allow us to handle any combination, however.
+
+## Summary
+
+We have recently had a new use case presented to us that involves a networking
+environment where multicast traffic is not allowed. This is problematic for the
+existing mDNS-based internal DNS system that is used for baremetal deployments.
+However, the targeted environment will have the capability to provide the
+necessary DNS records from an external DNS system, so we should be able to
+just disable the mDNS services in such an environment.
+
+## Motivation
+
+This enables the deployment of OpenShift on baremetal in environments where
+multicast is not supported.
+
+### Goals
+
+Make it possible to deploy OpenShift on baremetal without the mDNS services
+enabled.
+
+### Non-Goals
+
+An alternate method for automatically managing the necessary DNS records.
+In environments where mDNS is disabled, the deployer will be responsible for
+providing the necessary DNS records themselves.
+
+## Proposal
+
+Add a new configuration option to the baremetal install-config called
+`useMDNS`. This option will take a string value that specifies where the mDNS
+services should be deployed. Initially, this is expected to be one of
+"all", "none", or "master". This value will be populated into the baremetal
+platform status and consumed by machine-config-operator to determine which
+baremetal infrastructure services it should deploy on the nodes.
+
+### User Stories [optional]
+
+#### Story 1
+
+As a customer deploying OpenShift on baremetal in a networking environment
+where multicast is not allowed, I want to be able to disable the mDNS services
+that won't function there and provide DNS records myself.
```

Comment:
> I want to be able to disable the mDNS services
that won't function there and provide DNS records myself.

the ipi flow requirements is that it should be single interaction from user.. how/where does the user setup the DNS?

Reply:
How do we make sure the user has setup the DNS correctly and reduce the engineering effort in trying to debug incorrect setup?